safehttp/response_writer.go

```diff
@@ -25,18 +25,22 @@ type ResponseWriter struct {
 
 	// Having this field unexported is essential for security. Otherwise one can
 	// easily overwrite the struct bypassing all our safety guarantees.
-	header  Header
-	written bool
+	header           Header
+	appliedInterceps []AppliedInterceptor
+	req              *IncomingRequest
```

Comment:
Why is this a part of a `ResponseWriter` now?

Reply:
Well, both `appliedInterceps` and `req` needed to be part of the `ResponseWriter` for our design where the `ResponseWriter` would run the `commitPhase`. In the newer design I'm working on now the `ResponseWriter` still needs to own the `req` but instead of `appliedInterceps` it has a reference to the `handlerWithInterceptors` and the `commitPhase` is moved to the `handlerWithInterceptors`.